### PR TITLE
[MM-32476] Apps in the Marketplace

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This PR adds the needed types for Apps in the Marketplace. It also consolidates the typed used in the Marketplace in `src/types/marketplace.ts`.

I added a `AppManifest` type with only the needed fields for my PR. Should we go ahead and add all types?

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32476
https://mattermost.atlassian.net/browse/MM-32535

### CODE REVIEW

**Comment by hanzei** [comment]
@catalintomai Please review. I can't add you as a reviewer.

File: src/client/client4.ts [hunk 1]
```diff
@@ -3102,7 +3105,7 @@ export default class Client4 {
     };
 
     getMarketplacePlugins = (filter: string, localOnly = false) => {
-        return this.doFetch<MarketplacePlugin>(
+        return this.doFetch<MarketplacePlugin[]>(
```
Comment (hanzei) [inline src/client/client4.ts:3108]: Fixed this while touching the file

**Review by larkox — APPROVED** [review]

**Review by catalintomai — APPROVED** [review]

File: src/types/marketplace.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+export enum AuthorType {
+    Mattermost = 'mattermost',
+    Partner = 'partner',
+    Community = 'community',
+}
+
+export enum ReleaseStage {
+    Production = 'production',
```
Comment (catalintomai) [inline src/types/marketplace.ts:26]: TODO: add a documentation ticket to detail what do these mean in the apps context?

Comment (hanzei) [reply src/types/marketplace.ts:26]: @catalintomai These fields will be used in the new Marketplace design. What kind of documentation are you referring to?

**Pull request merged**